### PR TITLE
Problem: no explanation of reward configuration parameters 

### PR DESCRIPTION
**Pull request by lezzokafka**
Solution: 
Rewrote the section  "Validator Rewards": 
- The first part (overview) come with easy examples with graphs and for the layman, 
- Then we go more in-depth with the actual calculation and explain what's the network parameters with examples in the monetary expansion section.

- Took away the "Punishments" as it is now a separated one under #89

### CODE REVIEW

File: docs/getting-started/staking.md [hunk 1]
```diff
@@ -89,25 +89,51 @@ pub struct RewardsPoolState {
...
+- **S**: The total amount of staking, and
+- **R**: The reward rate per annum.
+
+Specifically, this reward rate `R` can be expressed by the following function:
 
 ```
-rewards of validator = total rewards * number of blocks proposed by the validator / total number of blocks
+R = (R0 / 1000) * exp(-S / tau)
 ```
+To virtualize this, if we set `tau=10 Billion`, `R0=350`, we have the following graph of the function:
```
Comment (devashishdxt) [inline docs/getting-started/staking.md]: virtualize?

Comment (lezzokafka) [reply docs/getting-started/staking.md]: Opps, visualize is what I mean

File: docs/getting-started/staking.md [hunk 2]
```diff
@@ -116,46 +142,65 @@ Configuration section in `genesis.json` for this:
...
-The number of tokens being released at each period is defined as:
+At the end of each reward epoch, the number of tokens being released at each period is defined as:
 
 ```
-R0 = rewards_config["monetary_expansion_r0"]
-tau = rewards_config["monetary_expansion_tau"]
-P = rewards_config["reward_period_seconds"]
+    R0 = rewards_config["monetary_expansion_r0"]
+    tau = rewards_config["monetary_expansion_tau"]
+    P = rewards_config["distribution_period:"]
```
Comment (devashishdxt) [inline docs/getting-started/staking.md]:
```suggestion
    P = rewards_config["distribution_period"]
```

File: docs/getting-started/staking.md [hunk 3]
```diff
@@ -89,25 +89,51 @@ pub struct RewardsPoolState {
...
 
-The recording of block proposer is done in `BeginBlock` right before rewards distribution.
+Note that the above reward rate is per annum, and the number of tokens being released to the reward pool (**N**) for that epoch is calculated by 
+
+```
+N = S  *  ( (1 + R) ^ (1 / f) - 1 )
+``` 
 
-#### Monetary expansion
+where `S` is the total amount of tokens staked by validators to participate in the consensus process; `R0` is the upper bound for the reward rate; `tau` is a time-dependent variable that controls the exceptional rate; `f` is the frequency of reward being distributed per year.
```
Comment (tomtau) [inline docs/getting-started/staking.md]: "exceptional rate" -> "exponential rate"?

File: docs/getting-started/staking.md [hunk 4]
```diff
@@ -116,46 +142,65 @@ Configuration section in `genesis.json` for this:
...
-R0 = rewards_config["monetary_expansion_r0"]
-tau = rewards_config["monetary_expansion_tau"]
-P = rewards_config["reward_period_seconds"]
+    R0 = rewards_config["monetary_expansion_r0"]
+    tau = rewards_config["monetary_expansion_tau"]
+    P = rewards_config["distribution_period:"]
 
-S = total_staking  # example: 1500000000_00000000
-Y = 365 * 24 * 60 * 60  # seconds of a year
+    S = total_staking  # example: 1500000000_00000000
```
Comment (devashishdxt) [inline docs/getting-started/staking.md]: Can you specify how this `total_staking` value is calculated for a rewards period?

**Review by tomtau — CHANGES_REQUESTED** [review]
examples are good -- it'll be all good to explain how setting individual parameters higher or lower will affect the emission.

File: docs/getting-started/staking.md (hunk 3 above)
Comment (lezzokafka) [reply docs/getting-started/staking.md]: Nice catch! Thanks Tomas

Comment (lezzokafka) [reply docs/getting-started/staking.md]: > examples are good -- it'll be all good to explain how setting individual parameters higher or lower will affect the emission.

Sure, I plan to put this in the next PR when resolving #80.

File: docs/getting-started/staking.md (hunk 4 above)
Comment (lezzokafka) [reply docs/getting-started/staking.md]: @devashishdxt  I change it to a description rather then a number here: 

    # total bonded amount of the active validators 
    at the end of the reward epoch
    S = total_staking

File: docs/getting-started/staking.md (hunk 2 above)
Comment (lezzokafka) [reply docs/getting-started/staking.md]: Thanks :)

**Pull request merged**